src/test/scala/org/apache/spark/sql/delta/DeleteSuiteBase.scala

```diff
@@ -0,0 +1,279 @@
+          (java.sql.Date.valueOf("2099-12-31"), 4))
+          .toDF("key", "value"), partitions)
+
+      checkDelete(Some("CURRENT_DATE > key"),
+        Row(java.sql.Date.valueOf("2099-12-31"), 4) :: Nil)
+      checkDelete(Some("CURRENT_DATE <= key"), Nil)
+    }
+  }
+
+  test("condition having current_timestamp - Partition by Timestamp") {
+    append(
+      Seq((java.sql.Timestamp.valueOf("2012-12-31 16:00:10.011"), 2),
+        (java.sql.Timestamp.valueOf("2099-12-31 16:00:10.011"), 4))
+        .toDF("key", "value"), Seq("key"))
+
+    checkDelete(Some("CURRENT_TIMESTAMP > key"),
+      Row(java.sql.Timestamp.valueOf("2099-12-31 16:00:10.011"), 4) :: Nil)
+    checkDelete(Some("CURRENT_TIMESTAMP <= key"), Nil)
+  }
+
+  Seq(true, false).foreach { isPartitioned =>
+    test(s"foldable condition - Partition=$isPartitioned") {
+      val partitions = if (isPartitioned) "key" :: Nil else Nil
+      append(Seq((2, 2), (1, 4), (1, 1), (0, 3)).toDF("key", "value"), partitions)
+
+      val allRows = Row(2, 2) :: Row(1, 4) :: Row(1, 1) :: Row(0, 3) :: Nil
+
+      checkDelete(Some("false"), allRows)
+      checkDelete(Some("1 <> 1"), allRows)
+      checkDelete(Some("1 > null"), allRows)
+      checkDelete(Some("true"), Nil)
+      checkDelete(Some("1 = 1"), Nil)
+    }
+  }
+
+  test("SC-12232: should not delete the rows where condition evaluates to null") {
+    append(Seq(("a", null), ("b", null), ("c", "v"), ("d", "vv")).toDF("key", "value").coalesce(1))
+
+    // "null = null" evaluates to null
+    checkDelete(Some("value = null"),
+      Row("a", null) :: Row("b", null) :: Row("c", "v") :: Row("d", "vv") :: Nil)
+
+    // these expressions evaluate to null when value is null
+    checkDelete(Some("value = 'v'"),
+      Row("a", null) :: Row("b", null) :: Row("d", "vv") :: Nil)
+    checkDelete(Some("value <> 'v'"),
+      Row("a", null) :: Row("b", null) :: Nil)
+  }
+
+  test("SC-12232: delete rows with null values using isNull") {
+    append(Seq(("a", null), ("b", null), ("c", "v"), ("d", "vv")).toDF("key", "value").coalesce(1))
+
+    // when value is null, this expression evaluates to true
+    checkDelete(Some("value is null"),
+      Row("c", "v") :: Row("d", "vv") :: Nil)
+  }
+
+  test("SC-12232: delete rows with null values using EqualNullSafe") {
+    append(Seq(("a", null), ("b", null), ("c", "v"), ("d", "vv")).toDF("key", "value").coalesce(1))
+
+    // when value is null, this expression evaluates to true
+    checkDelete(Some("value <=> null"),
+      Row("c", "v") :: Row("d", "vv") :: Nil)
+  }
+
+  test("do not support subquery test") {
+    append(Seq((2, 2), (1, 4), (1, 1), (0, 3)).toDF("key", "value"))
+    Seq((2, 2), (1, 4), (1, 1), (0, 3)).toDF("c", "d").createOrReplaceTempView("source")
+    val e = intercept[AnalysisException] {
+      executeDelete(target = s"delta.`$tempPath`", "key < (SELECT max(c) FROM source)")
+    }.getMessage
+    assert(e.contains("Subqueries are not supported"))
+  }
```

Comment:
Add Java API tests.

Reply:
For java tests need more work on enabling JUnit in SBT, we should do this in a separate PR.